tests/test_diaForcedSource.py

```diff
@@ -0,0 +1,207 @@
+        self.metadata = dafBase.PropertySet()
+
+        self.metadata.set("SIMPLE", "T")
+        self.metadata.set("BITPIX", -32)
+        self.metadata.set("NAXIS", 2)
+        self.metadata.set("NAXIS1", 1024)
+        self.metadata.set("NAXIS2", 1153)
+        self.metadata.set("RADECSYS", 'FK5')
+        self.metadata.set("EQUINOX", 2000.)
+
+        self.metadata.setDouble("CRVAL1", 215.604025685476)
+        self.metadata.setDouble("CRVAL2", 53.1595451514076)
+        self.metadata.setDouble("CRPIX1", 1109.99981456774)
+        self.metadata.setDouble("CRPIX2", 560.018167811613)
+        self.metadata.set("CTYPE1", 'RA---SIN')
+        self.metadata.set("CTYPE2", 'DEC--SIN')
+
+        self.metadata.setDouble("CD1_1", 5.10808596133527E-05)
+        self.metadata.setDouble("CD1_2", 1.85579539217196E-07)
+        self.metadata.setDouble("CD2_2", -5.10281493481982E-05)
+        self.metadata.setDouble("CD2_1", -8.27440751733828E-07)
+
+        self.wcs = afwGeom.makeSkyWcs(self.metadata)
+
+        self.flux0 = 10000
+        self.flux0_err = 100
+        self.exposureId = 1234
+        self.exposureTime = 200.
+        self.imageSize = [1024, 1153]
+        self.dateTime = "2014-05-13T17:00:00.000000000"
+
+        # Make images  with one source in them and distinct values and
+        # variance for each image.
+        # Direct Image
+        ms = afwImage.MaskedImageF(lsst.geom.ExtentI(self.imageSize[0] + 1,
+                                                     self.imageSize[1] + 1))
+        ms.image[100, 100, afwImage.LOCAL] = 10
+        ms.getVariance().set(1)
+        bbox = lsst.geom.BoxI(
+            lsst.geom.PointI(1, 1),
+            lsst.geom.ExtentI(self.imageSize[0],
+                              self.imageSize[1]))
+        self.mi = afwImage.MaskedImageF(ms, bbox, afwImage.LOCAL)
+        self.exposure = afwImage.makeExposure(self.mi, self.wcs)
+
+        detector = DetectorWrapper(
+            id=23, bbox=self.exposure.getBBox()).detector
+        visit = afwImage.VisitInfo(
+            exposureId=self.exposureId,
+            exposureTime=self.exposureTime,
+            date=dafBase.DateTime(self.dateTime,
+                                  dafBase.DateTime.Timescale.TAI))
+        self.exposure.setDetector(detector)
+        self.exposure.getInfo().setVisitInfo(visit)
+        self.exposure.setFilter(afwImage.Filter('g'))
+        self.exposure.getCalib().setFluxMag0((self.flux0, self.flux0_err))
+
+        # Difference Image
+        ms = afwImage.MaskedImageF(lsst.geom.ExtentI(self.imageSize[0] + 1,
+                                                     self.imageSize[1] + 1))
+        ms.image[100, 100, afwImage.LOCAL] = 20
+        ms.getVariance().set(2)
+        bbox = lsst.geom.BoxI(
+            lsst.geom.PointI(1, 1),
+            lsst.geom.ExtentI(self.imageSize[0],
+                              self.imageSize[1]))
+        self.mi = afwImage.MaskedImageF(ms, bbox, afwImage.LOCAL)
```

Comment:
Does `self.mi` need to be saved as a class attribute? You're just overwriting the value you set it to a few lines back, which suggests you don't need it outside of setting up `self.exposure` and `self.diffim`.